File: testdata/jennies/builders/basic_struct_defaults/PHPBuilder/src/BasicStructDefaults/SomeStructBuilder.php
<?php

namespace Grafana\Foundation\BasicStructDefaults;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BasicStructDefaults\SomeStruct>
 */
class SomeStructBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\BasicStructDefaults\SomeStruct $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\BasicStructDefaults\SomeStruct();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\BasicStructDefaults\SomeStruct
     */
    public function build()
    {
        return $this->internal;
    }

    public function id(int $id): static
    {
        $this->internal->id = $id;
    
        return $this;
    }

    public function uid(string $uid): static
    {
        $this->internal->uid = $uid;
    
        return $this;
    }

    /**
     * @param array<string> $tags
     */
    public function tags(array $tags): static
    {
        $this->internal->tags = $tags;
    
        return $this;
    }

    public function liveNow(bool $liveNow): static
    {
        $this->internal->liveNow = $liveNow;
    
        return $this;
    }

}

File: testdata/jennies/builders/basic_struct_defaults/CSharpBuilder/src/Grafana/Foundation/BasicStructDefaults/SomeStructBuilder.cs
namespace Grafana.Foundation.BasicStructDefaults;


public class SomeStructBuilder : Cog.IBuilder<SomeStruct>
{
    protected readonly SomeStruct @internal;

    public SomeStructBuilder()
    {
        this.@internal = new SomeStruct();
    }

    public SomeStructBuilder Id(long id)
    {
        this.@internal.Id = id;
        return this;
    }

    public SomeStructBuilder Uid(string uid)
    {
        this.@internal.Uid = uid;
        return this;
    }

    public SomeStructBuilder Tags(List<string> tags)
    {
        this.@internal.Tags = tags;
        return this;
    }

    public SomeStructBuilder LiveNow(bool liveNow)
    {
        this.@internal.LiveNow = liveNow;
        return this;
    }

    public SomeStruct Build()
    {
        return this.@internal;
    }
}
